Dockerfile

```diff
@@ -3,36 +3,42 @@ FROM ruby:2.6.6-alpine AS dev-build
 ARG DEV_PACKAGES="gcc libc-dev make yarn postgresql-dev build-base libxml2-dev libxslt-dev"
 
 WORKDIR /teacher-vacancy
+
+RUN apk add --no-cache libxml2 libxslt libpq tzdata nodejs $DEV_PACKAGES
+RUN echo "Europe/London" > /etc/timezone && \
+        cp /usr/share/zoneinfo/Europe/London /etc/localtime
+RUN gem install bundler:2.1.4 --no-document
+
+COPY Gemfile* ./
+RUN bundle install --no-binstubs --retry=5 --jobs=4 --no-cache --without development test
+
+COPY package.json yarn.lock ./
+RUN yarn install --check-files
+
 COPY . .
 
-RUN apk add --no-cache libxml2 libxslt libpq tzdata nodejs && \
-        apk add --no-cache --virtual .gem-installdeps $DEV_PACKAGES && \
-        gem install bundler:2.1.4 --no-document && \
-        bundle -v && \
-        bundle install --no-binstubs --retry=5 --jobs=4 --no-cache --without development test && \
-        echo "Europe/London" > /etc/timezone && \
-        cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
-        yarn install --check-files && \
-        RAILS_ENV=staging bundle exec rake webpacker:compile
+RUN RAILS_ENV=staging bundle exec rake webpacker:compile
```

Comment:
Why are you setting RAILS_ENV to staging for the generic Dockerfile?

Reply:
This is for the compilation of assets. It needs to run in any rails environment, but development has more dependencies.
staging is sufficient.